rpc/types/src/base.rs

```diff
@@ -10,76 +10,30 @@
 //! - <https://github.com/monero-project/monero/commit/2899379791b7542e4eb920b5d9d58cf232806937>
 //! - <https://github.com/monero-project/monero/issues/8722>
 //! - <https://github.com/monero-project/monero/pull/8843>
+//!
+//! This library doesn't contain [`rpc_access_request_base`](https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L114-L122)
+//! found in `monerod` as the type is practically deprecated.
+//!
+//! Although, [`AccessResponseBase`] still exists as to allow
+//! outputting the same JSON fields as `monerod` (even if deprecated).
```

Comment:
There is a `rpc_access_request_base` defined in `monerod` that pairs with the `rpc_access_response_base`.

This contains a `std::string: client` field, but since this is deprecated, I removed the implementation as we'll never use it - however - I kept `AccessResponseBase` as this actually affects the JSON output (i.e. it adds `top_hash: String`, `credits: u64`).

At the end of the day, the response is deprecated too but I thought following `monerod`'s JSON output more than not would be good. We can also remove this if wanted, they're always dead fields anyway.

Reply:
Added back `AccessRequestBase`.

FYI the empty bases (`Request`, `Response`) are special-cased by the macro since flattening an empty struct causes some errors.